src/uncrustify.cpp

```diff
@@ -312,10 +312,11 @@ static void redir_stdout(const char *output_file)
          LOG_FMT(LERR, "Unable to open %s for write: %s (%d)\n",
                  output_file, strerror(errno), errno);
          cpd.error_count++;
-         usage_exit(nullptr, nullptr, EX_IOERR);
+         return(usage(nullptr, nullptr, EX_IOERR));
       }
       LOG_FMT(LNOTE, "Redirecting output to %s\n", output_file);
    }
+   return(0);
```

Comment:
`EXIT_SUCCESS`

Reply:
Before I make this change, a word of explanation...

This function gets called like `if (int error = redir_stdout(...))`, so it's relying on it to return `0` if nothing goes wrong. Now, it *does* happen that `EXIT_SUCCESS==0`, but we'd be relying on that (as currently written, we *also* rely on other codes being non-zero), and it might not be as obvious that getting here won't trip the `if`.

That said, how would you feel about changing the return type from `int` to:

```c++
class ExitCode {
  ExitCode(int c) : code{c} {}
  operator int() { return code; }
  operator bool() { return code != EXIT_SUCCESS; }
protected:
  int code;
};
```

In practice, this is probably superfluous, but it insulates us against the possibility that `EXIT_SUCCESS != 0`, and would more clearly express the intent of the return value from `redir_stdout`.

Reply:
Hum... yeah cool, but I think `EXIT_SUCCESS` (and `EXIT_FAILURE`) are defined in the standard which will make it hard to break our assumption.

Reply:
Okay, I'll leave it alone for now. I don't think this would break anyway, except maybe on some really exotic platforms, but I made the consumers use `auto` so it will be that much easier if we decide at some point to wrap it. I also stuck a `[[nodiscard]]` on the function for good measure.